libs/agno/agno/team/team.py

```diff
@@ -6840,27 +6820,120 @@ def get_messages_for_session(
 
     def get_session_summary(self, session_id: Optional[str] = None, user_id: Optional[str] = None):
         """Get the session summary for the given session ID and user ID."""
-        if self.memory is None:
+        if self.team_session is None:
             return None
 
         session_id = session_id if session_id is not None else self.session_id
         if session_id is None:
             raise ValueError("Session ID is required")
 
-        # TODO: Add a db call to get the session summary
         return self.team_session.get_session_summary(session_id=session_id, user_id=user_id)
 
-    def get_user_memories(self, user_id: Optional[str] = None):
+    def get_user_memories(self, user_id: Optional[str] = None) -> Optional[List[UserMemory]]:
         """Get the user memories for the given user ID."""
-        if self.memory is None:
+        if self.memory_manager is None:
             return None
         user_id = user_id if user_id is not None else self.user_id
         if user_id is None:
             user_id = "default"
 
-        if isinstance(self.memory, Memory):
-            return self.memory.get_user_memories(user_id=user_id)
-        raise ValueError(f"Memory type {type(self.memory)} not supported")
+        return self.memory_manager.get_user_memories(user_id=user_id)
+
+    def add_interaction_to_team_context(
+        self, session_id: str, member_name: str, task: str, run_response: Union[RunResponse, TeamRunResponse]
+    ) -> None:
+        if self._team_context is None:
+            self._team_context = {}
+        if session_id not in self._team_context:
+            self._team_context[session_id] = {}
+        self._team_context[session_id] = {
```

Comment:
This will blow up memory. I think we only need to store this for the current run, so it can be reset at the start of each run.